src/Controller/Component/AuthComponent.php

```diff
@@ -147,7 +154,8 @@ class AuthComponent extends Component
         'logoutRedirect' => null,
         'authError' => null,
         'unauthorizedRedirect' => true,
-        'storage' => 'Session'
+        'storage' => 'Session',
+        'earlyAuth' => false
```

Comment:
@jeremyharris I don't think that would be easy as allowing any event would require using implementedEvents() and events get registered as soon as component instance is created before `Controller::initialize()` is called.